### PR TITLE
[RW-6481]: Support bypass RAS link and fix some RAS linking issues

### PR DESCRIPTION
**Pull request by yonghaoy**
Description:
Support bypassing RAS Link, UI is fully feature protected. 
Also after RAS unblock our development work, I am able to fully test this feature, and fix some issues based on feedback:

1. Email is not unique & immutable. We should use prefrred_name. 
2. I use content, not contentEncoding when passing user info.
3. IAL status's url is varies, those part is not needed, only the part should be enough and valid in acr.  

Tested locally and working now

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/admin/admin-user-bypass.tsx [hunk 1]
```diff
@@ -50,6 +50,7 @@ const getBypassedModules = (user: AdminTableUser): Array<AccessModule> => {
     ...(user.dataUseAgreementBypassTime ? [AccessModule.DATAUSEAGREEMENT] : []),
     ...(user.eraCommonsBypassTime ? [AccessModule.ERACOMMONS] : []),
     ...(user.twoFactorAuthBypassTime ? [AccessModule.TWOFACTORAUTH] : []),
+    ...(user.rasLinkLoginGovBypassTime ? [AccessModule.RASLINKLOGINGOV] : []),
```
Comment (calbach) [inline ui/src/app/pages/admin/admin-user-bypass.tsx:53]: You can also conditionally add this to the registration dashboard test-only "bypass all" flow: https://github.com/all-of-us/workbench/blob/81a6213a125eee09ea537ecbb78dfe74961a6bcc/ui/src/app/pages/homepage/registration-dashboard.tsx#L316

File: api/src/main/java/org/pmiops/workbench/ras/RasLinkService.java [hunk 2]
```diff
@@ -128,7 +126,7 @@ public DbUser linkRasLoginGovAccount(String authCode, String redirectUrl) {
 
   /** Validates user has IAL2 setup. See class javadoc Step2 for more details. */
   static boolean isIal2(String acrClaim) {
-    return Arrays.asList(acrClaim.split(" ")).contains(ACR_CLAIM_IAL_2_IDENTIFIER);
+    return acrClaim.contains(ACR_CLAIM_IAL_2_IDENTIFIER);
```
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/ras/RasLinkService.java:129]: Did the acr format change? Is there an updated doc link we can add to this service file?

File: api/src/main/java/org/pmiops/workbench/ras/RasLinkConstants.java [hunk 3]
```diff
@@ -27,14 +27,11 @@ private RasLinkConstants() {}
...
 
-  // The login.gov's email field which we choose as login.gov username.
-  static final String EMAIl_FIELD_NAME = "email";
-
   // The identifier that indicate users login using login.gov account.
   static final String LOGIN_GOV_IDENTIFIER_LOWER_CASE = "@login.gov";
 
   // The string in ACR claim that we can use to identify user IAL status.
-  static final String ACR_CLAIM_IAL_2_IDENTIFIER = "https://stsstg.nih.gov/assurance/ial/2";
+  static final String ACR_CLAIM_IAL_2_IDENTIFIER = "/assurance/ial/2";
```
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/ras/RasLinkConstants.java:34]: Is this the exact value? Or are we just matching a suffix now?

File: api/src/main/resources/workbench-api.yaml [hunk 4]
```diff
@@ -5286,6 +5287,12 @@ definitions:
       eraCommonsCompletionTime:
         type: integer
         format: int64
+      rasLinkLoginGovBypassTime:
```
Comment (calbach) [inline api/src/main/resources/workbench-api.yaml:5290]: Are these being populated now? I don't see the change in this PR - is it happening automatically in a mapper somewhere?

File: api/src/main/java/org/pmiops/workbench/ras/RasLinkService.java (hunk 2 above)
Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/ras/RasLinkService.java:129]: The acr format does not change. I misunderstood that before. I didn't find standard doc to explains this and I feel ther is no standard for IAL status. What I know is RAS use this to represent IAL status, and the format is signing server `url + "/assurance/ial/2" ` I didn't ask if signing server url is always be the rasHost, but seems `/assurance/ial/2` is a good represent. So I just simplify this to see if that contains '/assurance/ial/2'

File: api/src/main/java/org/pmiops/workbench/ras/RasLinkConstants.java (hunk 3 above)
Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/ras/RasLinkConstants.java:34]: It is `https://stsstg.nih.gov/assurance/ial/2 https://stsstg.nih.gov/assurance/aal2`. Explained in previous comment, matching might be engough?. Do you think we should do the exact match or suffix match? 
I saw few different RAS urls prefix, so little bit concerns if the prefix can alwasy match, hence choose this suffiix way...

File: api/src/main/resources/workbench-api.yaml (hunk 4 above)
Comment (yonghaoy) [reply api/src/main/resources/workbench-api.yaml:5290]: I did update the query and a new getAasLinkLoginGovBypassTime() in userDao's `DbAdminTableUser`. And it seems to be working.

File: api/src/main/java/org/pmiops/workbench/ras/RasLinkConstants.java (hunk 3 above)
Comment (calbach) [reply api/src/main/java/org/pmiops/workbench/ras/RasLinkConstants.java:34]: Without having documentation from RAS on this, my default preference would be full match - partial match could potentially be exploited depending on where this value is coming from.

However, if you feel the RAS API is unstable here and we don't have good documentation from them yet, I'm fine with the partial suffix match for now (but I would probably have a ticket to come back and update this ahead of launch).

**Review by calbach — APPROVED** [review]

File: api/src/main/java/org/pmiops/workbench/ras/RasLinkConstants.java (hunk 3 above)
Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/ras/RasLinkConstants.java:34]: I do have their documents and available in google doc. But they ask to delete from Google. 
It is in their box account: https://app.box.com/file/751989029656?s=wnjs4a0nqavhjwy3wdu5nv1vwlkx7dmk
Screen shot on IAL and AAL intro in this claim:
https://screenshot.googleplex.com/A2hx9t42Y5DeKRd

**Pull request merged**